From stable-bounces@linux.kernel.org Sat Apr  7 16:45:03 2007
From: Stephen Hemminger <shemminger@linux-foundation.org>
Date: Sat, 07 Apr 2007 16:42:06 -0700
Subject: skge: turn carrier off when down
Cc: netdev@vger.kernel.org, stable@kernel.org
Message-ID: <20070407234235.566060329@linux-foundation.org>
Content-Disposition: inline; filename=skge-carrier.patch

From: Stephen Hemminger <shemminger@linux-foundation.org>

Driver needs to turn off carrier when down, otherwise it can
confuse bonding and bridging and looks like carrier is on immediately
when it is brought back up.

Signed-off-by: Stephen Hemminger <shemminger@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/net/skge.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/net/skge.c
+++ b/drivers/net/skge.c
@@ -2462,6 +2462,7 @@ static int skge_down(struct net_device *
 		printk(KERN_INFO PFX "%s: disabling interface\n", dev->name);
 
 	netif_stop_queue(dev);
+	netif_carrier_off(dev);
 	if (hw->chip_id == CHIP_ID_GENESIS && hw->phy_type == SK_PHY_XMAC)
 		cancel_rearming_delayed_work(&skge->link_thread);
 
